Description: add parrot template
Last-Update: 2017-03-19

--- /dev/null
+++ calamares-3.1/src/branding/parrot/branding.desc
@@ -0,0 +1,27 @@
+---
+componentName:  parrot
+
+strings:
+    productName:         Parrot GNU/Linux
+    shortProductName:    Parrot
+    version:             3.6
+    shortVersion:        3.6
+    versionedName:       Parrot GNU/Linux 3.6 "testing"
+    shortVersionedName:  Parrot 3.6
+    bootloaderEntryName: Parrot
+    productUrl:          https://www.parrotsec.org/
+    supportUrl:          https://community.parrotsec.org/
+    knownIssuesUrl:      https://community.parrotsec.org
+    releaseNotesUrl:     https://blog.parrotsec.org
+
+images:
+    productLogo:         "squid.png"
+    productIcon:         "squid.png"
+    productWelcome:      "languages.png"
+
+slideshow:               "show.qml"
+
+style:
+   sidebarBackground:    "#292F34"
+   sidebarText:          "#FFFFFF"
+   sidebarTextSelect:    "#292F34"
--- /dev/null
+++ calamares-3.1/src/branding/parrot/show.qml
@@ -0,0 +1,63 @@
+/* === This file is part of Calamares - <http://github.com/calamares> ===
+ *
+ *   Copyright 2015, Teo Mrnjavac <teo@kde.org>
+ *
+ *   Calamares is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   Calamares is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.0;
+import calamares.slideshow 1.0;
+
+Presentation
+{
+    id: presentation
+
+    Timer {
+        interval: 5000
+        running: false
+        repeat: true
+        onTriggered: presentation.goToNextSlide()
+    }
+    
+    Slide {
+
+        Image {
+            id: background
+            source: "squid.png"
+            width: 200; height: 200
+            fillMode: Image.PreserveAspectFit
+            anchors.centerIn: parent
+        }
+        Text {
+            anchors.horizontalCenter: background.horizontalCenter
+            anchors.top: background.bottom
+            text: "This is a customizable QML slideshow.<br/>"+
+                  "Distributions should provide their own slideshow and list it in <br/>"+
+                  "their custom branding.desc file.<br/>"+
+                  "To create a Calamares presentation in QML, import calamares.slideshow,<br/>"+
+                  "define a Presentation element with as many Slide elements as needed."
+            wrapMode: Text.WordWrap
+            width: root.width
+            horizontalAlignment: Text.Center
+        }
+    }
+
+    Slide {
+        centeredText: "This is a second Slide element."
+    }
+
+    Slide {
+        centeredText: "This is a third Slide element."
+    }
+}
