Description: additional modules and customizations
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2017-03-19

--- calamares-3.1.orig/calamares.desktop
+++ calamares-3.1/calamares.desktop
@@ -1,13 +1,12 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Calamares
-GenericName=System Installer
-Keywords=calamares;system;installer
-TryExec=calamares
-Exec=pkexec /usr/bin/calamares
-Comment=Calamares — System Installer
-Icon=calamares
+Name=Parrot Installer
+GenericName=Parrot Installer
+TryExec=gksu calamares
+Exec=gksu /usr/bin/calamares
+Comment=System Installer
+Icon=gdeb
 Terminal=false
-StartupNotify=true
+StartupNotify=false
 Categories=Qt;System;
--- calamares-3.1.orig/settings.conf
+++ calamares-3.1/settings.conf
@@ -80,20 +80,23 @@ sequence:
   - locale
   - keyboard
   - localecfg
-#  - luksbootkeyfile
+  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
+#  - initcpiocfg
+#  - initcpio
   - users
   - displaymanager
   - networkcfg
   - hwclock
   - services
 #  - dracut
+  - grubefiinstall
+  - packages
+  - initramfscfg
   - initramfs
-#  - grubcfg
+  - grubcfg
   - bootloader
   - umount
 - show:
@@ -109,7 +112,7 @@ sequence:
 # Only the name of the branding component (directory) should be specified here, Calamares
 # then takes care of finding it and loading the contents.
 # YAML: string.
-branding: default
+branding: parrot
 
 # If this is set to true, Calamares will show an "Are you sure?" prompt right before
 # each execution phase, i.e. at points of no return. If this is set to false, no prompt
--- calamares-3.1.orig/src/modules/fstab/fstab.conf
+++ calamares-3.1/src/modules/fstab/fstab.conf
@@ -1,13 +1,12 @@
 ---
 mountOptions:
     default: defaults,noatime
-    btrfs: defaults,noatime,space_cache,autodefrag
+    btrfs: defaults,noatime,nodiratime,space_cache,autodefrag,compress=lzo
 ssdExtraMountOptions:
     ext4: discard
     jfs: discard
     xfs: discard
     swap: discard
     btrfs: discard,compress=lzo
-crypttabOptions: luks
+crypttabOptions: luks,keyscript=/bin/cat
 # For Debian and Debian-based distributions, change the above line to:
-# crypttabOptions: luks,keyscript=/bin/cat
--- /dev/null
+++ calamares-3.1/src/modules/grubefiinstall/bootloader-eficonfig
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# Replaced apt installation with dpkg (offline) by pavroo
+# Last update 2017/03/02
+
+DEVCHROOT=$(mount | grep proc | grep calamares | awk '{print $3}' | sed -e "s#/proc##g")
+
+if [ -d /sys/firmware/efi/efivars ]; then
+    echo " * Replacing grub-pc with grub-efi..."
+    # grub-probe doesn't work in a CHROOT...
+    mv $DEVCHROOT/usr/sbin/grub-probe $DEVCHROOT/usr/sbin/grub-probe.bak
+    echo "exit 0" > $DEVCHROOT/usr/sbin/grub-probe
+    chmod +x $DEVCHROOT/usr/sbin/grub-probe
+    # chroot $DEVCHROOT apt-get --allow-unauthenticated -y install grub-efi
+    mkdir -p $DEVCHROOT/boot/efi/EFI/parrot
+    mkdir -p $DEVCHROOT/debs
+    cp /lib/live/mount/medium/offline/* $DEVCHROOT/debs/
+    #chroot $DEVCHROOT dpkg -i /debs/*.deb
+
+cat > $DEVCHROOT/tmp/efi-install << FOO
+#!/bin/bash
+cd debs
+dpkg -i *.deb
+cd ..
+FOO
+
+    chmod +x $DEVCHROOT/tmp/efi-install
+    chroot $DEVCHROOT /tmp/efi-install
+    rm -f $DEVCHROOT/tmp/efi-install
+    rm -rf $DEVCHROOT/debs
+    rm -f $DEVCHROOT/usr/sbin/grub-probe
+    mv $DEVCHROOT/usr/sbin/grub-probe.bak $DEVCHROOT/usr/sbin/grub-probe
+fi
--- /dev/null
+++ calamares-3.1/src/modules/grubefiinstall/module.desc
@@ -0,0 +1,6 @@
+---
+type:       "job"
+name:       "grubefiinstall"
+interface:  "process"
+command:    "./bootloader-eficonfig"
+timeout:    600
--- calamares-3.1.orig/src/modules/initramfs/main.py
+++ calamares-3.1/src/modules/initramfs/main.py
@@ -26,7 +26,7 @@ def run():
 
     :return:
     """
-    return_code = target_env_call(["update-initramfs", "-k", "all", "-u"])
+    return_code = target_env_call(["update-initramfs", "-t", "-u", "-k", "all"])
 
     if return_code != 0:
         return "Failed to run update-initramfs on the target", "The exit code was {}".format(return_code)
--- calamares-3.1.orig/src/modules/locale/locale.conf
+++ calamares-3.1/src/modules/locale/locale.conf
@@ -3,5 +3,5 @@ region:                     "America"
 zone:                       "New_York"
 
 # GeoIP settings. Leave commented out to disable GeoIP.
-#localeGenPath:             "/etc/locale.gen"
-#geoipUrl:                  "freegeoip.net"
+localeGenPath:             "/etc/locale.gen"
+geoipUrl:                  "freegeoip.net"
--- calamares-3.1.orig/src/modules/packages/packages.conf
+++ calamares-3.1/src/modules/packages/packages.conf
@@ -11,7 +11,7 @@
 #  - portage	 - Gentoo package manager
 #  - entropy	 - Sabayon package manager
 #
-backend: packagekit
+backend: apt
 
 update_db: true
 
@@ -29,20 +29,8 @@ update_db: true
 # storage called "packageOperations" and it is processed
 # after the static list in the job configuration.
 #
-#operations:
-#  - install:
-#      - pkg1
-#      - pkg2
-#  - remove:
-#      - pkg3
-#      - pkg4
-#  - try_install:   # no system install failure if a package cannot be installed
-#      - pkg5
-#  - try_remove:    # no system install failure if a package cannot be removed
-#      - pkg2
-#      - pkg1
-#  - install:
-#      - pkgs6
-#      - pkg7
-#  - localInstall:
-#      - /path/to/pkg8
+operations:
+  - remove:
+      - '^live-*'
+      - calamares
+      - memtest86+
--- calamares-3.1.orig/src/modules/unpackfs/main.py
+++ calamares-3.1/src/modules/unpackfs/main.py
@@ -244,12 +244,9 @@ def run():
     to rootMountPoint, e.g.:
     configuration:
         unpack:
-            - source: "/path/to/filesystem.img"
+            - source: "/lib/live/mount/rootfs/filesystem.squashfs"
               sourcefs: "ext4"
               destination: ""
-            - source: "/path/to/another/filesystem.sqfs"
-              sourcefs: "squashfs"
-              destination: ""
 
     :return:
     """
--- calamares-3.1.orig/src/modules/unpackfs/unpackfs.conf
+++ calamares-3.1/src/modules/unpackfs/unpackfs.conf
@@ -1,8 +1,5 @@
 ---
 unpack:
-    -   source: "/path/to/filesystem.img"
+    -   source: "/lib/live/mount/rootfs/filesystem.squashfs"
         sourcefs: "ext4"
         destination: ""
-    -   source: "/path/to/another/filesystem.sqfs"
-        sourcefs: "squashfs"
-        destination: ""
--- calamares-3.1.orig/src/modules/users/users.conf
+++ calamares-3.1/src/modules/users/users.conf
@@ -1,17 +1,24 @@
 ---
 defaultGroups:
-    - users
-    - lp
-    - video
-    - network
-    - storage
-    - wheel
     - audio
+    - bluetooth
+    - cdrom
+    - dialout
+    - dip
+    - floppy
+    - lpadmin
+    - netdev
+    - plugdev
+    - powerdev
+    - scanner
+    - sudo
+    - video
+	- debian-tor
+	- libvirt
+	- fuse
+
 autologinGroup:  autologin
 doAutologin:     true
-
-# remove the following line to avoid creating /etc/sudoers.d/10-installer
-sudoersGroup:    wheel
-
+sudoersGroup:    sudo
 setRootPassword: true
 doReusePassword: true
