Description: Fix bootloader module
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-01-21

Index: parrot-installer-3.1.8/src/modules/bootloader/main.py
===================================================================
--- parrot-installer-3.1.8.orig/src/modules/bootloader/main.py
+++ parrot-installer-3.1.8/src/modules/bootloader/main.py
@@ -212,18 +212,18 @@ def install_grub(efi_directory, fw_type)
     :param fw_type:
     """
     if fw_type == "efi":
-       try:
-           check_target_env_call(["dpkg",
+        try:
+            check_target_env_call(["dpkg",
                                "-i",
                                "--force-all",
                                "/lib/live/mount/medium/pool/main/g/grub2/grub*-common*.deb"])
-           check_target_env_call(["dpkg",
+            check_target_env_call(["dpkg",
                                "-i",
                                "--force-all",
                                "/lib/live/mount/medium/pool/main/g/grub2/grub-efi*.deb"])
         except Exception:
-           pass
-       
+            pass
+
         print("Bootloader: grub (efi)")
         install_path = libcalamares.globalstorage.value("rootMountPoint")
         install_efi_directory = install_path + efi_directory
@@ -294,17 +294,18 @@ def install_grub(efi_directory, fw_type)
         if boot_loader["installPath"] is None:
             return
         
-       try:
-           check_target_env_call(["dpkg",
+        try:
+            check_target_env_call(["dpkg",
                                "-i",
                                "--force-all",
                                "/lib/live/mount/medium/pool/main/g/grub2/grub*-common*.deb"])
-           check_target_env_call(["dpkg",
+            check_target_env_call(["dpkg",
                                "-i",
                                "--force-all",
                                "/lib/live/mount/medium/pool/main/g/grub2/grub-pc*.deb"])
         except Exception:
-           pass
+            pass
+
         check_target_env_call([libcalamares.job.configuration["grubInstall"],
                                "--target=i386-pc",
                                "--recheck",
