Description: Parrot base Patches
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2017-12-19

--- parrot-installer-3.1.8.orig/calamares.desktop
+++ parrot-installer-3.1.8/calamares.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Calamares
+Name=Parrot Installer
 GenericName=System Installer
 Keywords=calamares;system;installer
 TryExec=calamares
@@ -11,105 +11,3 @@ Icon=calamares
 Terminal=false
 StartupNotify=true
 Categories=Qt;System;
-
-
-Name[ca]=Calamares
-Icon[ca]=calamares
-GenericName[ca]=Instal·lador de sistema
-Comment[ca]=Calamares — Instal·lador de sistema
-Name[da]=Calamares
-Icon[da]=calamares
-GenericName[da]=Systeminstallationsprogram
-Comment[da]=Calamares — Systeminstallationsprogram
-Name[de]=Calamares
-Icon[de]=calamares
-GenericName[de]=Installation des Betriebssystems
-Comment[de]=Calamares - Installation des Betriebssystems
-Name[en_GB]=Calamares
-Icon[en_GB]=calamares
-GenericName[en_GB]=System Installer
-Comment[en_GB]=Calamares — System Installer
-Name[es]=Calamares
-Icon[es]=calamares
-GenericName[es]=Instalador del Sistema
-Comment[es]=Calamares — Instalador del Sistema
-Name[fr]=Calamares
-Icon[fr]=calamares
-GenericName[fr]=Installateur système
-Comment[fr]=Calamares - Installateur système
-Name[he]=קלמארס
-Icon[he]=קלמארס
-GenericName[he]=אשף התקנה
-Comment[he]=קלמארס - אשף התקנה
-Name[hr]=Calamares
-Icon[hr]=calamares
-GenericName[hr]=Instalacija sustava
-Comment[hr]=Calamares — Instalacija sustava
-Name[hu]=Calamares
-Icon[hu]=calamares
-GenericName[hu]=Rendszer Telepítő
-Comment[hu]=Calamares — Rendszer Telepítő
-Name[id]=Calamares
-Icon[id]=calamares
-GenericName[id]=Pemasang
-Comment[id]=Calamares — Pemasang Sistem
-Name[is]=Calamares
-Icon[is]=calamares
-GenericName[is]=Kerfis uppsetning
-Comment[is]=Calamares — Kerfis uppsetning
-Name[ja]=Calamares
-Icon[ja]=calamares
-GenericName[ja]=システムインストーラー
-Comment[ja]=Calamares — システムインストーラー
-Name[lt]=Calamares
-Icon[lt]=calamares
-GenericName[lt]=Sistemos diegimas į kompiuterį
-Comment[lt]=Calamares — sistemos diegyklė
-Name[nl]=Calamares
-Icon[nl]=calamares
-GenericName[nl]=Installatieprogramma
-Comment[nl]=Calamares — Installatieprogramma
-Name[pl]=Calamares
-Icon[pl]=calamares
-GenericName[pl]=Instalator systemu
-Comment[pl]=Calamares — Instalator systemu
-Name[pt_BR]=Calamares
-Icon[pt_BR]=calamares
-GenericName[pt_BR]=Instalador de Sistema
-Comment[pt_BR]=Calamares — Instalador de Sistema
-Name[cs_CZ]=Calamares
-Icon[cs_CZ]=calamares
-GenericName[cs_CZ]=Instalační program systému
-Comment[cs_CZ]=Calamares - instalační program systému
-Name[ru]=Calamares
-Icon[ru]=calamares
-GenericName[ru]=Установщик системы
-Comment[ru]=Calamares - Установщик системы
-Name[sk]=Calamares
-Icon[sk]=calamares
-GenericName[sk]=Inštalátor systému
-Comment[sk]=Calamares — Inštalátor systému
-Name[sv]=Calamares
-Icon[sv]=calamares
-GenericName[sv]=Systeminstallerare
-Comment[sv]=Calamares — Systeminstallerare
-Name[zh_CN]=Calamares
-Icon[zh_CN]=calamares
-GenericName[zh_CN]=系统安装程序
-Comment[zh_CN]=Calamares — 系统安装程序
-Name[zh_TW]=Calamares
-Icon[zh_TW]=calamares
-GenericName[zh_TW]=系統安裝程式
-Comment[zh_TW]=Calamares ── 系統安裝程式
-Name[ast]=Calamares
-Icon[ast]=calamares
-GenericName[ast]=Instalador del sistema
-Comment[ast]=Calamares — Instalador del sistema
-Name[pt_PT]=Calamares
-Icon[pt_PT]=calamares
-GenericName[pt_PT]=Instalador de Sistema
-Comment[pt_PT]=Calamares - Instalador de Sistema
-Name[tr_TR]=Calamares
-Icon[tr_TR]=calamares
-GenericName[tr_TR]=Sistem Yükleyici
-Comment[tr_TR]=Calamares — Sistem Yükleyici
--- parrot-installer-3.1.8.orig/settings.conf
+++ parrot-installer-3.1.8/settings.conf
@@ -84,8 +84,8 @@ sequence:
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
+#  - initcpiocfg
+#  - initcpio
   - users
   - displaymanager
   - networkcfg
@@ -109,7 +109,7 @@ sequence:
 # Only the name of the branding component (directory) should be specified here, Calamares
 # then takes care of finding it and loading the contents.
 # YAML: string.
-branding: default
+branding: parrot
 
 # If this is set to true, Calamares will show an "Are you sure?" prompt right before
 # each execution phase, i.e. at points of no return. If this is set to false, no prompt
--- /dev/null
+++ parrot-installer-3.1.8/src/branding/parrot/branding.desc
@@ -0,0 +1,27 @@
+---
+componentName:  parrot
+
+strings:
+    productName:         Parrot GNU/Linux
+    shortProductName:    Parrot
+    version:             3.10
+    shortVersion:        3.10
+    versionedName:       Parrot 3.10 (Intruder)
+    shortVersionedName:  Parrot 3.10
+    bootloaderEntryName: Parrot
+    productUrl:          https://www.parrotsec.org
+    supportUrl:          https://community.parrotsec.org
+    knownIssuesUrl:      https://dev.parrotsec.org/groups/parrot/issues
+    releaseNotesUrl:     https://blog.parrotsec.org
+
+images:
+    productLogo:         "logo.jpg"
+    productIcon:         "logo.jpg"
+    productWelcome:      "languages.png"
+
+slideshow:               "show.qml"
+
+style:
+   sidebarBackground:    "#292F34"
+   sidebarText:          "#FFFFFF"
+   sidebarTextSelect:    "#292F34"
--- /dev/null
+++ parrot-installer-3.1.8/src/branding/parrot/show.qml
@@ -0,0 +1,117 @@
+/*
+ * Parrot theme for Calamares
+ * 
+ * Copyright 2017 Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
+ * 
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ * 
+ * 
+ */
+
+import QtQuick 2.0;
+import calamares.slideshow 1.0;
+
+Presentation
+{
+    id: presentation
+
+    Timer {
+        interval: 5000
+        running: false
+        repeat: true
+        onTriggered: presentation.goToNextSlide()
+    }
+    
+    Slide {
+
+        Image {
+            id: background
+            source: "slides.png"
+            width: 450; height: 159
+            fillMode: Image.PreserveAspectFit
+            anchors.centerIn: parent
+        }
+        Text {
+            anchors.horizontalCenter: background.horizontalCenter
+            anchors.top: background.bottom
+            text: "This is the installer of the Parrot GNU/Linux Distribution.<br/>"+
+                  "It includes a full portable laboratory for security"+
+                  "and digital forensics experts, but it also includes"+
+                  "all you need to develop your own softwares or protect"+
+                  "your privacy with anonymity and crypto tools."
+            wrapMode: Text.WordWrap
+            width: root.width
+            horizontalAlignment: Text.Center
+        }
+    }
+
+    Slide {
+
+        Image {
+            id: background
+            source: "links.jpg"
+            width: 300; height: 140
+            fillMode: Image.PreserveAspectFit
+            anchors.centerIn: parent
+        }
+        Text {
+            anchors.horizontalCenter: background.horizontalCenter
+            anchors.top: background.bottom
+            text: "<h3>Security</h3><br>Parrot Security includes a full arsenal of security oriented tools to perform penetration tests, security audits and more. With a Parrot usb drive in your pocket you will always be sure to have all you need with you."
+            wrapMode: Text.WordWrap
+            width: root.width
+            horizontalAlignment: Text.Center
+        }
+    }
+
+    Slide {
+
+        Image {
+            id: background
+            source: "fingerprint.jpg"
+            width: 300; height: 147
+            fillMode: Image.PreserveAspectFit
+            anchors.centerIn: parent
+        }
+        Text {
+            anchors.horizontalCenter: background.horizontalCenter
+            anchors.top: background.bottom
+            text: "<h3>Privacy</h3><br> Parrot includes by default TOR, I2P, anonsurf, gpg, zulucrypt, veracrypt, truecrypt, luks and many other technologies designed to defend your privacy and your identity."
+            wrapMode: Text.WordWrap
+            width: root.width
+            horizontalAlignment: Text.Center
+        }
+    }
+    
+    Slide {
+
+        Image {
+            id: background
+            source: "devel.jpg"
+            width: 360; height: 122
+            fillMode: Image.PreserveAspectFit
+            anchors.centerIn: parent
+        }
+        Text {
+            anchors.horizontalCenter: background.horizontalCenter
+            anchors.top: background.bottom
+            text: "<h3>Development</h3><br> If you need a comfortable environment with updated frameworks and useful libraries already installed, Parrot will amaze you as it includes a full development-oriented environment with some powerful editors and IDEs pre-installed and many other tools installable from our repository."
+            wrapMode: Text.WordWrap
+            width: root.width
+            horizontalAlignment: Text.Center
+        }
+    }
+}
--- parrot-installer-3.1.8.orig/src/modules/bootloader/bootloader.conf
+++ parrot-installer-3.1.8/src/modules/bootloader/bootloader.conf
@@ -5,13 +5,13 @@ efiBootLoader: "grub"
 
 # systemd-boot configuration files settings, set kernel and initramfs file names
 # and amount of time before default selection boots
-kernel: "/vmlinuz-linux"
-img: "/initramfs-linux.img"
-fallback: "/initramfs-linux-fallback.img"
-timeout: "10"
+kernel: "/vmlinuz"
+img: "/initrd.img"
+fallback: "/initrd.img.old"
+timeout: "5"
 # Optionally set the menu entry name and kernel name to use in systemd-boot.
 # If not specified here, these settings will be taken from branding.desc.
-# bootloaderEntryName: "Generic GNU/Linux"
+bootloaderEntryName: "Parrot GNU/Linux"
 # kernelLine: ", with Stable-Kernel"
 # fallbackKernelLine:  ", with Stable-Kernel (fallback initramfs)"
 
--- parrot-installer-3.1.8.orig/src/modules/bootloader/main.py
+++ parrot-installer-3.1.8/src/modules/bootloader/main.py
@@ -212,6 +212,18 @@ def install_grub(efi_directory, fw_type)
     :param fw_type:
     """
     if fw_type == "efi":
+       try:
+           check_target_env_call(["dpkg",
+                               "-i",
+                               "--force-all",
+                               "/lib/live/mount/medium/pool/main/g/grub2/grub*-common*.deb"])
+           check_target_env_call(["dpkg",
+                               "-i",
+                               "--force-all",
+                               "/lib/live/mount/medium/pool/main/g/grub2/grub-efi*.deb"])
+        except Exception:
+           pass
+       
         print("Bootloader: grub (efi)")
         install_path = libcalamares.globalstorage.value("rootMountPoint")
         install_efi_directory = install_path + efi_directory
@@ -281,7 +293,18 @@ def install_grub(efi_directory, fw_type)
         boot_loader = libcalamares.globalstorage.value("bootLoader")
         if boot_loader["installPath"] is None:
             return
-
+        
+       try:
+           check_target_env_call(["dpkg",
+                               "-i",
+                               "--force-all",
+                               "/lib/live/mount/medium/pool/main/g/grub2/grub*-common*.deb"])
+           check_target_env_call(["dpkg",
+                               "-i",
+                               "--force-all",
+                               "/lib/live/mount/medium/pool/main/g/grub2/grub-pc*.deb"])
+        except Exception:
+           pass
         check_target_env_call([libcalamares.job.configuration["grubInstall"],
                                "--target=i386-pc",
                                "--recheck",
--- parrot-installer-3.1.8.orig/src/modules/fstab/fstab.conf
+++ parrot-installer-3.1.8/src/modules/fstab/fstab.conf
@@ -1,13 +1,13 @@
 ---
 mountOptions:
     default: defaults,noatime
-    btrfs: defaults,noatime,space_cache,autodefrag
+    btrfs: defaults,noatime,nodiratime,relatime,space_cache,autodefrag,compress
 ssdExtraMountOptions:
     ext4: discard
     jfs: discard
     xfs: discard
     swap: discard
-    btrfs: discard,compress=lzo
-crypttabOptions: luks
+    btrfs: discard
+crypttabOptions: luks,keyscript=/bin/cat
 # For Debian and Debian-based distributions, change the above line to:
 # crypttabOptions: luks,keyscript=/bin/cat
--- parrot-installer-3.1.8.orig/src/modules/mount/mount.conf
+++ parrot-installer-3.1.8/src/modules/mount/mount.conf
@@ -15,6 +15,12 @@ extraMounts:
     - device: /run/udev
       mountPoint: /run/udev
       options: bind
+    - device: /lib/live/
+      mountPoint: /lib/live/
+      options: bind
+    - device: /lib/live/mount/medium
+      mountPoint: /lib/live/mount/medium
+      options: bind
 
 extraMountsEfi:
     - device: efivarfs
--- parrot-installer-3.1.8.orig/src/modules/packages/packages.conf
+++ parrot-installer-3.1.8/src/modules/packages/packages.conf
@@ -12,12 +12,12 @@
 #  - entropy     - Sabayon package manager
 #  - dummy       - Dummy manager, only logs
 #
-backend: dummy
+backend: apt
 
 # If set to true, a package-manager specific update procedure
 # is run first (only if there is internet) to update the list
 # of packages and dependencies.
-update_db: true
+update_db: false
 
 #
 # List of maps with package operations such as install or remove.
@@ -112,11 +112,6 @@ update_db: true
 # of packages, this can lead to a considerable time savings.
 #
 operations:
-  - install:
-    - vi
-    - wget
-    - binutils
-  - remove:
-    - vi
-    - wget
-    - binutils
+  - try_remove:
+    - '^live-*'
+    - parrot-installer
--- parrot-installer-3.1.8.orig/src/modules/partition/partition.conf
+++ parrot-installer-3.1.8/src/modules/partition/partition.conf
@@ -21,7 +21,7 @@ ensureSuspendToDisk:    true
 neverCreateSwap:        false
 
 # Correctly draw nested (e.g. logical) partitions as such.
-drawNestedPartitions:   false
+drawNestedPartitions:   true
 
 # Show/hide partition labels on manual partitioning page.
 alwaysShowPartitionLabels: true
--- parrot-installer-3.1.8.orig/src/modules/services/services.conf
+++ parrot-installer-3.1.8/src/modules/services/services.conf
@@ -5,16 +5,11 @@ services:
   - name: "NetworkManager"  #name of the service file
     mandatory: false        #true=> if enabling fails the installer errors out and quits
                             #false=>if enabling fails print warning to console and continue
-  - name: "cups"
-    mandatory: false
 
 targets:
   - name: "graphical"
-    mandatory: true
-
-disable:
-  - name: "pacman-init"
     mandatory: false
 
+disable: []
 # Example to express an empty list:
 # disable: []
--- parrot-installer-3.1.8.orig/src/modules/unpackfs/unpackfs.conf
+++ parrot-installer-3.1.8/src/modules/unpackfs/unpackfs.conf
@@ -33,6 +33,6 @@ unpack:
 
 # You can list filesystem source paths relative to the Calamares run
 # directory, if you use -d (this is only useful for testing, though).
-    -   source: ./example.sqfs
+    -   source: "/lib/live/mount/medium/live/filesystem.squashfs"
         sourcefs: squashfs
         destination: ""
--- parrot-installer-3.1.8.orig/src/modules/users/users.conf
+++ parrot-installer-3.1.8/src/modules/users/users.conf
@@ -19,8 +19,19 @@ defaultGroups:
     - storage
     - wheel
     - audio
+    - cdrom
+    - dip
+    - floppy
+    - plugdev
+    - powerdev
+    - netdev
+    - scanner
+    - bluetooth
+    - fuse
+    - sudo
+    - libvirt
 autologinGroup:  autologin
-doAutologin:     true
+doAutologin:     false
 
 # remove the following line to avoid creating /etc/sudoers.d/10-installer
 sudoersGroup:    wheel
